docs/pipeline.rst

```diff
@@ -197,3 +197,141 @@ to optimize the conversion by giving the converter additional
 information, even if the model to convert is included in a
 pipeline. That why the option mechanism was implemented:
 :ref:`l-conv-options`.
+
+Investigate discrepencies
+=========================
+
+A wrong converter may introduce introduce discrepencies
+in a converter pipeline but it is not alway easy to
+isolate the source of the differences. The function
+:func:`collect_intermediate_steps
+<skl2onnx.helpers.collect_intermediate_steps>`
+may then be used to investigate each component independently.
+The following piece of code is extracted from unit test
+`test_investigate.py <https://github.com/onnx/sklearn-onnx/
+blob/master/tests/test_investigate.py>`_ and converts
+a pipeline and each of its components independently.
+
+::
+
+    import numpy
+    from numpy.testing import assert_almost_equal
+    from sklearn.pipeline import Pipeline
+    from sklearn.preprocessing import StandardScaler
+    import onnxruntime
+    from skl2onnx.helpers import collect_intermediate_steps, compare_objects
+
+    # Let's fit a model.
+    data = numpy.array([[0, 0], [0, 0], [2, 1], [2, 1]],
+                       dtype=numpy.float32)
+    model = Pipeline([("scaler1", StandardScaler()),
+                      ("scaler2", StandardScaler())])
+    model.fit(data)
+
+    # Convert and collect every operator in a pipeline
+    # and modifies the current pipeline to keep
+    # intermediate inputs and outputs when method
+    # predict or transform is called.
+    steps = collect_intermediate_steps(model, "pipeline",
+                                       [("input",
+                                         FloatTensorType([1, 2]))])
+
+    # Method and transform is called.
+    model.transform(data)
+
+    # Loop on every operator.
+    for step in steps:
+    
+        # The ONNX for this operator.
+        short_onnx = step['short_onnx']
+        
+        # Use onnxruntime to compute ONNX outputs
+        sess = onnxruntime.InferenceSession(short_onnx.SerializeToString())
+    
+        # Let's use the initial data as the ONNX model
+        # contains all nodes from the first inputs to this node.
+        onnx_outputs = sess.run(None, {'input': data})
+        onnx_output = onnx_outputs[0]
+        skl_outputs = step['model']._debug.outputs['transform']
+        
+        # Compares the outputs between scikit-learn and onnxruntime.
+        assert_almost_equal(onnx_output, skl_outputs)
+        
+        # A function which is able to deal with different types.
+        compare_objects(onnx_output, skl_outputs)
+
+Investigate missing converters
+==============================
+
+Many converters can be missing before converting a pipeline.
+Exception :class:`MissingShapeCalculator 
+<skl2onnx.common.exceptions.MissingShapeCalculator>` is
+raised when the first missing one is found.
+The previous snippet of code can be modified to find all of
+them.
+
```

Comment:
Wording suggestion:
The previous snippet of code can be modified to find all of them, as shown below.